### ISSUE TITLE
False positive unassigned variable

### ISSUE THREAD

**Issue by mhmxs** [uber-go/nilaway#207]
Something changed in the latest version of nilaway, because one of our existing code now has a detected issue.

```
go: downloading go.uber.org/nilaway v0.0.0-20240216175439-fb8b98c43554
```

Here is the code, which looks a valid error handling:

```golang
var awsErr smithy.APIError // Is an interface
return errors.As(err, &awsErr) && awsErr.ErrorCode() == "something"
```

Nilaway reports the following:

```
error: Potential nil panic detected. Observed nil flow from source to dereference point: 
        - rgw/lifecycleconfig_helpers.go:167:36: unassigned variable `awsErr` called `ErrorCode()`
```

**Comment by sonalmahajan15** [comment]
@mhmxs , thanks for reporting! We haven't yet modeled `errors.As` support in NilAway, so this is unfortunately a case of a known false positive (as tracked in issue #95 ). But the interesting bit here is that only after the recent changes did this error start showing up for you, which I am not sure why, since this error has been known to be reported since a long time. 🤔

**Comment by sonalmahajan15** [comment]
(Marking this as a duplicate of #95. Feel free to continue the discussion on that thread.)

**Comment by mhmxs** [comment]
> as tracked in issue #95

Thanks @sonalmahajan15 I was seeking "False positive" issues only before reporting :angel:
